### PR TITLE
Handle time scale FITS metadata in map.date

### PR DESCRIPTION
**Pull request by dstansby**
Section 9.2.1 https://fits.gsfc.nasa.gov/standard40/fits_standard40aa-le.pdf says there is a TIMESYS keyword, that can be set to denote the time scale of times in a FITS header. This PR:

- Sets the time scale from TIMESYS if present
- Explicitly sets the timescale to 'utc', which is the mandated default, if TIMESYS isn't present.

It looks like the timescale was implicitly set to UTC anyway by default (either by us or astropy?), so this shouldn't be a big issue, but we should still support alternative time scales from the FITS header.

### CODE REVIEW

File: sunpy/map/mapbase.py [hunk 1]
```diff
@@ -657,6 +657,16 @@ def date(self):
         This is taken from the 'DATE-OBS' FITS keyword.
         """
         time = self.meta.get('date-obs', None)
+        # Get the time scale if present
+        # SDO specifies the 'TAI' scale in their time string, so detect that
+        # and special case it (even though it's probably not FITS compliant)
+        if time is not None and 'TAI' in time:
+            timesys = 'TAI'
+            if 'timesys' in self.meta:
+                warnings.warn('Found "TAI" in time string, ignoring TIMESYS keyword')
```
Comment (Cadair) [inline sunpy/map/mapbase.py:666]: shouldn't we special case this in the sdo source?

**Comment by dstansby** [comment]
Note to self (discussed on call): we should fix the funny TAI date in SDO map source, and deprecate TAI parsing in `parse_time`.

**Comment by Cadair** [comment]
I am not sure we should deprecate it in `parse_time`, but not relying on it in Map would be good given the other changes in here.

File: sunpy/map/mapbase.py (hunk 1 above)
Comment (Cadair) [inline sunpy/map/mapbase.py:666]:
```suggestion
                warnings.warn('Found "TAI" in time string, ignoring TIMESYS keyword', SunpyUserWarning)
```

**Review by wtbarnes — APPROVED** [review]

**Review by Cadair — APPROVED** [review]

File: sunpy/map/mapbase.py [hunk 2]
```diff
@@ -657,6 +657,20 @@ def date(self):
...
+        # Get the time scale
+        if time is not None and 'TAI' in time:
+            # SDO specifies the 'TAI' scale in their time string, which is parsed
+            # by parse_time(). If a different timescale is also present, warn the
+            # user that it will be ignored.
+            timesys = 'TAI'
+            timesys_meta = self.meta.get('timesys', '').upper()
+            if timesys_meta != 'TAI':
+                warnings.warn('Found "TAI" in time string, ignoring TIMESYS keyword '
+                              f'which is set to "{timesys_meta}".')
```
Comment (Cadair) [inline sunpy/map/mapbase.py:669]:
```suggestion
                              f'which is set to "{timesys_meta}".', SunpyUserWarning)
```

File: changelog/4881.bugfix.rst [hunk 3]
```diff
@@ -0,0 +1,4 @@
+`sunpy.map.GenericMap.date` now has its time scale set from the 'TIMESYS' FITS keyword,
+if it is present. If it isn't present the time scale defaults to 'utc', which is unchanged
```
Comment (Cadair) [inline changelog/4881.bugfix.rst:2]:
```suggestion
if it is present. If it isn't present the time scale defaults to 'UTC', which is unchanged
```

**Review by Cadair — APPROVED** [review]

File: sunpy/map/mapbase.py [hunk 4]
```diff
@@ -657,6 +657,20 @@ def date(self):
         This is taken from the 'DATE-OBS' FITS keyword.
         """
         time = self.meta.get('date-obs', None)
+        # Get the time scale
+        if time is not None and 'TAI' in time:
+            # SDO specifies the 'TAI' scale in their time string, which is parsed
```
Comment (nabobalis) [inline sunpy/map/mapbase.py:662]: Stupid question, why was this not moved to the SDO map source? I can't find a comment or any reference.

**Review by nabobalis — APPROVED** [review]

File: sunpy/map/mapbase.py (hunk 4 above)
Comment (dstansby) [reply sunpy/map/mapbase.py:662]: Why was what not moved to the map source?

Comment (nabobalis) [reply sunpy/map/mapbase.py:662]: This code deals with SDO specific time handing no?

Comment (dstansby) [reply sunpy/map/mapbase.py:662]: Just in case there is other data out there that has this TAI timestamp, I think it's worth keeping it in here.

**Pull request merged**